pkg/controller/statefulset/stateful_set.go

```diff
@@ -437,7 +444,7 @@ func (ssc *StatefulSetController) worker(ctx context.Context) {
 func (ssc *StatefulSetController) sync(ctx context.Context, key string) error {
 	startTime := time.Now()
 	defer func() {
-		klog.V(4).Infof("Finished syncing statefulset %q (%v)", key, time.Since(startTime))
+		ssc.logger.V(4).Info("Finished syncing statefulset", "statefulSet", key, "time", time.Since(startTime))
```

Comment:
This is why I don't like objects that contain a logger: this function gets a context which might contain a logger which is different from ssc.logger, but it uses the cached logger instead.

Can you remove ssc.logger and just pass through a logger instance where needed (i.e. for the handler functions)?